### PR TITLE
Feat: support resource topology for endpoints

### PR DESCRIPTION
**Pull request by wonderflow**
Signed-off-by: Jianbo Sun <jianbo.sjb@alibaba-inc.com>

### Description of your changes

1. support resource topology for endpoints check
2. refactor the code to reduce duplicate

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/velaql/providers/query/endpoint.go [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+	"github.com/oam-dev/kubevela/pkg/utils"
+	querytypes "github.com/oam-dev/kubevela/pkg/velaql/providers/query/types"
+	wfContext "github.com/oam-dev/kubevela/pkg/workflow/context"
+	"github.com/oam-dev/kubevela/pkg/workflow/types"
+)
+
+// GeneratorServiceEndpoints generator service endpoints is available for common component type,
+// such as webservice or helm
+// it can not support the cloud service component currently
+func (h *provider) GeneratorServiceEndpoints(wfctx wfContext.Context, v *value.Value, act types.Action) error {
```
Comment (barnettZQG) [inline pkg/velaql/providers/query/endpoint.go:48]: Add some unit test cases?

File: pkg/stdlib/pkgs/query.cue [hunk 2]
```diff
@@ -50,6 +50,9 @@
 		deployVersion?:   string
 		revision?:        string
 		latest?:          bool
+		resourceTree?: {
```
Comment (barnettZQG) [inline pkg/stdlib/pkgs/query.cue:53]: Why add this field here?  `listAppliedResources` means only query the resources that render from the application.

File: pkg/velaql/providers/query/handler_test.go [hunk 3]
```diff
@@ -962,19 +974,18 @@ options: {
 			"http://text.example.com",
 			"10.10.10.10:81",
 			"text.example.com:81",
-			// helmRelease
-			fmt.Sprintf("http://%s:30002", gatewayIP),
```
Comment (barnettZQG) [inline pkg/velaql/providers/query/handler_test.go:966]: Why remove the rules from the helm release?

Comment (wonderflow) [reply pkg/velaql/providers/query/handler_test.go:966]: it not work in tests

File: pkg/stdlib/pkgs/query.cue (hunk 2 above)
Comment (wonderflow) [reply pkg/stdlib/pkgs/query.cue:53]: The struct was added by this value internally

**Review by Somefive — APPROVED** [review]

**Pull request merged**